addons/mrp/mrp.py

```diff
@@ -353,6 +353,14 @@ def onchange_product_tmpl_id(self, cr, uid, ids, product_tmpl_id, product_qty=0,
             }
         return res
 
+    def onchange_product_id(self, cr, uid, ids, product_id, context=None):
+        res = {}
+        if product_id:
+            res['value'] = {
+                'product_tmpl_id': self.pool["product.product"].browse(cr, uid, product_id, context).product_tmpl_id.id,
+            }
+        return res
```

Comment:
I suppose you're trying to update the product template of the BoM when created from Manufacturing Order. It seems it does not work. Why don't you have a hidden field on the manufacturing order view with the product template and pass it in as context for new BoMs?